src/Components/AddTodo/AddTodo.tsx

```diff
@@ -0,0 +1,140 @@
+import React from 'react';
+import './AddTodo.scss';
+
+import { User, Todo } from '../../types';
+
+interface Props {
+  users: User[];
+  todos: Todo[];
+  addTodo: (todo: Todo) => void,
+}
+
+interface State {
+  userId: number;
+  title: string;
+  titleError: string;
+  userError: string;
+}
+
+export class AddTodo extends React.Component<Props, State> {
+  state: State = {
+    userId: 0,
+    title: '',
+    titleError: '',
+    userError: '',
+  };
+
+  resetForm = () => {
+    this.setState({
+      userId: 0,
+      title: '',
+      titleError: '',
+      userError: '',
+    });
+  };
+
+  handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({
+      title: event.target.value,
+      titleError: '',
+    });
+  };
+
+  handleSelectChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    this.setState({
+      userId: +event.target.value,
+      userError: '',
+    });
+  };
+
+  handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+    const { userId, title } = this.state;
+
+    if (title.trim().length === 0) {
+      this.setState({ titleError: 'Please, enter valid title' });
+
+      return;
+    }
+
+    if (userId === 0) {
+      this.setState({ userError: 'Please, choose user' });
+
+      return;
+    }
+
+    const newTodo = {
+      userId,
+      id: this.props.todos.length + 1,
```

Comment:
this will only work if the ids match the indices. it might not always be the case